### DISCUSSION TITLE
Adding a time attribute to the CGNS file after converting Plot3D files

### DISCUSSION THREAD

**Discussion by loopless** [CGNS/CGNS#306] [CGNStalk]
I am using the CGNS utility plot3d_to_cgns utility to convert Plot3D CFD files to CGNS.  Generally the utility works great...

However, either the Plot3D files do not contain a "time" attribute or it is not converted. The time attribute is required by some third-party software I am using that reads CGNS. I was looking to modify the utility to add a command line parameter with the "time" and add that attribute to the generated CGNS file. I am a complete CGNS newbie and was hoping someone could help me out with some example code.

**Comment by MicK7** [comment]
The converter will add a Reference time in the CGNS file and not store it in a baseiterative data structure. This may explain why third party software do not detect a valid time.
Is the time stored in PLOT3D a valid time or should the user provide a more accurate value  ?

**Reply by loopless** [reply]
Yes, exactly. I see the time is written in the reference. As you note,  the third party is looking for the times in the baseiterative structure. So I modified the converter to put a time in a baseiterative structure. That almost worked.. however, the problem is that I think the third party is looking for ZoneIterativeData. As you can tell I am flailing around a bit!
I was going to change the converter to write one time step as per https://cgns.github.io/CGNS_docs_current/user/additional.html#sec:timedep

**Reply by MicK7** [reply]
Your modification seems quite good. Which third party are you using ?

**Reply by loopless** [reply]
I confirmed my change does create a CGNS file that the third party can read. I'm afraid I can't discuss  the name of the third-party.

**Reply by MicK7** [reply]
Cool. No worries about the name.
If the change is not too big do you think you can share it ?
